packages/celotool/src/lib/prometheus.ts

```diff
@@ -113,69 +101,100 @@ async function helmParameters(
       { context },
       fetchEnv(envVar.KUBERNETES_CLUSTER_ZONE)
     )
+    clusterName = getDynamicEnvVarValue(
+      DynamicEnvVar.KUBERNETES_CLUSTER_NAME,
+      { context },
+      clusterName
+    )
+    stackdriverDisabled = getDynamicEnvVarValue(
+      DynamicEnvVar.PROM_SIDECAR_DISABLED,
+      { context },
+      clusterName
+    )
   } else {
     gcloudProject = fetchEnv(envVar.TESTNET_PROJECT_NAME)
     gcloudRegion = fetchEnv(envVar.KUBERNETES_CLUSTER_ZONE)
+    stackdriverDisabled = fetchEnvOrFallback(envVar.PROMETHEUS_DISABLE_STACKDRIVER_SIDECAR, 'false')
   }
 
+  return [clusterName, gcloudProject, gcloudRegion, stackdriverDisabled]
+}
+
+function getRemoteWriteParameters(context?: string): string[] {
+  const remoteWriteUrl = getDynamicEnvVarValue(
+    DynamicEnvVar.PROM_REMOTE_WRITE_URL,
+    { context },
+    fetchEnv(envVar.PROMETHEUS_REMOTE_WRITE_URL)
+  )
+  const remoteWriteUser = getDynamicEnvVarValue(
+    DynamicEnvVar.PROM_REMOTE_WRITE_USERNAME,
+    { context },
+    fetchEnv(envVar.PROMETHEUS_REMOTE_WRITE_USERNAME)
+  )
+  const remoteWritePassword = getDynamicEnvVarValue(
+    DynamicEnvVar.PROM_REMOTE_WRITE_PASSWORD,
+    { context },
+    fetchEnv(envVar.PROMETHEUS_REMOTE_WRITE_PASSWORD)
+  )
+  return [
+    `--set remote_write.url='${remoteWriteUrl}'`,
+    `--set remote_write.basic_auth.username='${remoteWriteUser}'`,
+    `--set remote_write.basic_auth.password='${remoteWritePassword}'`,
+  ]
+}
+
+async function helmParameters(context?: string, clusterConfig?: BaseClusterConfig) {
+  const usingGCP = !clusterConfig || clusterConfig.cloudProvider === CloudProvider.GCP
+  const [clusterName, gcloudProject, gcloudRegion, stackdriverDisabled] = getK8sContextVars(
+    clusterConfig,
+    context
+  )
+  const cloudProvider = clusterConfig ? getCloudProviderPrefix(clusterConfig!) : 'gcp'
+
   const params = [
     `--set namespace=${kubeNamespace}`,
     `--set gcloud.project=${gcloudProject}`,
     `--set gcloud.region=${gcloudRegion}`,
     `--set prometheus.imageTag=${prometheusImageTag}`,
     `--set serviceAccount.name=${kubeServiceAccountName}`,
     `--set cluster=${clusterName}`,
+    `--set stackdriver_metrics_prefix=external.googleapis.com/prometheus/${clusterName}`,
```

Comment:
Could you move this down to logically group it with the rest of the stackdriver related settings?

Reply:
Done